UKHomeOffice/sgmr-service
# Create user email

TO TEST

- make sure you are signed out
- go to /register

----

**If required fields are empty the form will error**
- click submit without entering any data
> you should see error messages for firstname, lastname, mobileNumber, email, password

**If mobile field is not a number the form will error**
- enter first name, last name
- enter letters in mobile field
- click submit
> you should see error message for mobile number

**If email field is not a valid email format the form will error**
- enter firstname, lastname, and valid mobile number
- enter an invalid email address format
- click submit
> you should see error message for email address

**If confirm email does not match email the form will error**
- enter firstname, lastname, valid mobile, valid email
- leave confirm email blank and click submit
> you should see error message for confirm email
- enter something in confirm email that is different to email
- click submit
> you should see error message for confirm email

**If password less than 8 characters the form will error**
- enter firstname, lastname, valid mobile, valid email, valid confirmation email
- enter password of 7 or fewer characters
- click submit
> you should see error message for password

**If confirm password does not match password the form will error*
- enter firstname, lastname, valid mobile, valid email, valid confirm email, valid password
- leave password confirm blank
- click submit
> you should see error message for password confirm
- enter something in password confirm that does not match password
- click submit
> you should see error message for password confirm

----

**If email address already registered form will error**
- enter valid data for every field
- enter the email address as one you have previously registered
- click submit
> You should see a page level error that the user already is registered

----

**If there are no errors, user will be registered**
- enter valid data for every field
- click submit
> You should see a successful POST in network
> You should be taken to the verify page and asked to enter your verification code

----
